src/input/mod.rs

```diff
@@ -1574,6 +1475,133 @@ impl State {
         }
     }
 
+    /// Used with cursor triggered events to check whether the keyboard focus should be updated.
+    fn maybe_update_focus(
```

Comment:
I think I named it that because under was initially set to None, and it looked as though there were some branches where it was never set to a value. Since the focus doesn't change everytime I thought it best to prefix it with some kind of conditional

Reply:
would `cursor_focus_event` or something similar be a better name?

Reply:
changed it to `cursor_focus` since it's only called for cursor clicks and cursor following